### PR TITLE
Added SSL config for Kafka publisher

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Added SSL config for Kafka publisher

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@mohitdeuex

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json [hunk 1]
```diff
@@ -42,8 +42,38 @@
...
+      "description": "Kafka SSL keystore password",
+      "type": "string"
+    },
+    "SSLKeyPassword": {
+      "description": "Kafka SSL key password",
+      "type": "string"
     }
   },
-  "required": ["topics"],
+  "required": ["topics", "securityProtocol"],
```
Comment (mohitdeuex) [inline catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json:77]: let's make it optional, if someone doesn't want to configure , then it's fine

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java [hunk 2]
```diff
@@ -23,6 +25,19 @@ public class KafkaWebhookEventPublisher extends WebhookPublisher {
   public KafkaWebhookEventPublisher(Webhook webhook, CollectionDAO dao) {
     super(webhook, dao);
     this.webhook = webhook;
+    if (webhook.getKafkaProperties().getSecurityProtocol().equals("SSL")) {
+      // configuration for SSL Encryption
+      properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, webhook.getKafkaProperties().getSecurityProtocol());
```
Comment (mohitdeuex) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java:30]: make it optional

File: catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json (hunk 1 above)
Comment (parthp2107) [reply catalog-rest-service/src/main/resources/json/schema/configuration/kafkaEventConfiguration.json]: Addressed in the latest commit.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java (hunk 2 above)
Comment (parthp2107) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaWebhookEventPublisher.java]: Addressed in the latest commit.

**Review by mohitdeuex — APPROVED** [review]

**Pull request merged**